hping: add page (#3305)
tldr-pages/tldr: pages/common/hping.md

# hping

> > Hping is a command-line oriented TCP/IP packet assembler/analyzer. The interface is inspired by the ping UNIX command.

Reviewer: Perhaps this will read better?
```suggestion
> Command-line oriented TCP/IP packet assembler and analyzer.
> Inspired by the `ping` command.
```

> More information: <http://www.hping.org>.

> - Make a ping with TCP protocol on localhost:

Reviewer: Perhaps this will read better?
```suggestion
- Ping localhost over TCP:
```

`hping3 {{localhost}}`

> - Make a ping with TCP on 127.0.0.1 address on port 80:

Reviewer: Perhaps this will read better?
```suggestion
- Ping over TCP on a specific port:
```

Author: I propose to use the format you defined in the last comment:
`Ping an IP address over TCP on a specific port`

Reviewer: Sounds good to me! There are only suggestions after all.

`hping3 -p {{80}} -S {{12.0.0.1}}`

> - Make a ping with UDP on 127.0.0.1 address on port 80:

Reviewer: Perhaps this will read better?
```suggestion
- Ping an IP address over UDP on port 80:
```

`hping3 --udp -p {{80}} -S {{127.0.0.1}}`

> - Scan TCP port 80,3000,9000 on 127.0.0.1 address:

Reviewer: Perhaps this will read better?
```suggestion
- Scan a set of TCP ports on a specific IP address:
```

`hping3 --scan {{80,3000,9000}} -S {{127.0.0.1}}`

> - Make a charge test on 127.0.0.1 on port 80:

Reviewer: Perhaps this will read better?
```suggestion
- Perform a charge test against an IP address on port 80:
```

# hping

> Command-line oriented TCP/IP packet assembler and analyzer.
> Inspired by the `ping` command.
> More information: <http://www.hping.org>.

- Ping localhost over TCP:

`hping3 {{localhost}}`

- Ping an IP address over TCP on a specific port:

> `hping3 -p {{80}} -S {{12.0.0.1}}`

Reviewer 2: Let's change all the IPs to 192.168.1.1 so that they are something different and consistent with other pages.

- Ping an IP address over UDP on port 80:

`hping3 --udp -p {{80}} -S {{127.0.0.1}}`

- Scan a set of TCP ports on a specific IP address:

`hping3 --scan {{80,3000,9000}} -S {{127.0.0.1}}`

> - Perform a charge test against an IP address on port 80:

Reviewer 2: We can leave out "against an IP address". It has been mentioned multiple times in the previous descriptions and is understood by now.
